lib/multierror/multierror_test.go

```diff
@@ -0,0 +1,76 @@
+package multierror_test
+
+import (
+	"errors"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/multierror"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+var (
+	OneErr   = errors.New("err one")
+	TwoErr   = errors.New("err two")
+	ThreeErr = errors.New("err three")
+	FourErr  = errors.New("err four")
+	FiveErr  = errors.New("err four")
+	SixErr   = errors.New("err four")
+	SevenErr = errors.New("err four")
+)
+
+type nameError struct {
+	name string
+}
+
+func (e nameError) Error() string {
+	return fmt.Sprintf("checkEven: Given number %s is not an even number", e.name)
+}
+
+func TestSanityIsError(t *testing.T) {
+	err := multierror.Wrap(OneErr, ThreeErr, FourErr)
+	assert.True(t, errors.Is(err, OneErr))
+	assert.False(t, errors.Is(err, TwoErr))
+	assert.True(t, errors.Is(err, ThreeErr))
+	assert.True(t, errors.Is(err, FourErr))
+	realErrStrings := []string{OneErr.Error(), ThreeErr.Error(), FourErr.Error()}
+	assert.Equal(t, strings.Join(realErrStrings, multierror.Separator), err.Error())
```

Comment:
This last assert is simply a change detector - it reimplements `MultiError.Error` such that if the impl of the latter changes, the test will break.

If we expect `.Error()` to contain the error strings of all errors, better would be to assert that `err.Error()` contains substrings `OneErr.Error()`, `TwoError.Error()`, etc.

Reply:
You're right, changed testing to reflect this